cmd/state-installer/cmd.go

```diff
@@ -237,39 +244,21 @@ func execute(out output.Outputer, cfg *config.Instance, an analytics.Dispatcher,
 		}
 	}
 
-	// Detect state tool alongside installer executable
-	installerPath := filepath.Dir(osutils.Executable())
-	packagedStateExe := filepath.Join(installerPath, installation.BinDirName, constants.StateCmd+exeutils.Extension)
-
-	// Detect whether this is a fresh install or an update
-	isUpdate := false
-	switch {
-	case (params.sourceInstaller == "install.sh" || params.sourceInstaller == "install.ps1") && fileutils.FileExists(packagedStateExe):
-		logging.Debug("Not using update flow as installing via " + params.sourceInstaller)
-		params.sourcePath = installerPath
-		break
-	case params.force:
-		logging.Debug("Not using update flow as --force was passed")
-		break // When ran with `--force` we always use the install UX
-	case params.sourcePath == "" && fileutils.FileExists(packagedStateExe):
-		// Facilitate older versions of state tool which do not invoke the installer with `--source-path`
-		logging.Debug("Using update flow as installer is alongside payload")
-		isUpdate = true
-		params.sourcePath = installerPath
-	case stateToolInstalled:
-		// This should trigger AFTER the check above where sourcePath is defined
-		logging.Debug("Using update flow as state tool is already installed")
-		isUpdate = true
```

Comment:
This is also dropping cases of `params.sourcePath = installerPath`, have you verified that this is no longer needed?

Reply:
Good point. There is no case where we _don't_ want to have `params.sourcePath` set so I've updated the code to reflect that.